### PR TITLE
Clock system tff 176

### PR DESCRIPTION
**Pull request by Cafferoq**
### Summary

The clock module that can be mocked out for time based tests and also will trigger the notification emails.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Ensure tests work
2. Ensure clock init code makes sense. Part if commented out as it waits to integrate with the email system.

### Links

- [TFF-176](https://msoese.atlassian.net/browse/TFF-176)

### CODE REVIEW

File: src/lib/clock.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+setTimeout(() => {
+    //TODO: connect email system here
+    //email Students
+    //setInterval(/*Email Students*/, 86400000);
+}, (moment.tz(Date.now(), "America/Chicago").hour(16).minute(30).second(0).format('X')
+ - moment.tz(Date.now(), "America/Chicago").format('X')) * 1000);
+
+export function isBeforeNow(timestamp){
+    let currentTime = moment.tz(Date.now(), "America/Chicago");
+    return timestamp < parseInt(currentTime.format('X'));
```
Comment (thofmann) [inline src/lib/clock.js]: Why not just use ~`return timestamp < Date.now() / 1000`?

Moment seems like overkill for this function, and timezones are completely unnecessary when using unix timestamps.

File: src/lib/clock.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import moment from 'moment-timezone';
+
+//initialization code
+setTimeout(() => {
+    //TODO: connect email system here
+    //email Students
+    //setInterval(/*Email Students*/, 86400000);
+}, (moment.tz(Date.now(), "America/Chicago").hour(16).minute(30).second(0).format('X')
```
Comment (thofmann) [inline src/lib/clock.js]: What happens if the timeout is created after 5:30pm?

File: test/unit/lib/clock.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { isBeforeNow } from '../../../.dist/lib/clock';
+import { assert } from 'chai';
+
+describe("Clock module", () => {
+    it("knows 0 is before now", () => {
+        assert.isTrue(isBeforeNow(0));
+    });
+
+    it("knows the future is not before now", () => {
```
Comment (thofmann) [inline test/unit/lib/clock.js]: ただしいます

File: test/unit/lib/clock.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { isBeforeNow } from '../../../.dist/lib/clock';
+import { assert } from 'chai';
+
+describe("Clock module", () => {
+    it("knows 0 is before now", () => {
```
Comment (thofmann) [inline test/unit/lib/clock.js]: ただしいます

File: src/lib/clock.js (hunk 1 above)
Comment (Cafferoq) [inline src/lib/clock.js]: At the time I couldn't remember if it was just 1000 to divide. That's why I converted it the way I did. Will change this.

File: src/lib/clock.js (hunk 2 above)
Comment (Cafferoq) [inline src/lib/clock.js]: Will fix

File: src/lib/clock.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+setTimeout(() => {
+    //TODO: connect email system here
+    //email Students
+    //setInterval(/*Email Students*/, 86400000);
+}, getInitialTimeoutTime();
+
+export function isBeforeNow(timestamp){
+    return timestamp < Date.now() / 1000);
```
Comment (thofmann) [inline src/lib/clock.js]: Extra `)` at the end?

**Pull request merged**